docs/stored-id-handling.md

```diff
@@ -0,0 +1,254 @@
+    Where the data is stored. Accepted values are `'post_meta'` or `'post_content'`.
+
+  - **`attributes`** (`array`):
+    Additional details depending on the location:
+    - **For post meta:**
+      - `meta_key` (`string`): The post meta key. **(Required)**
+    - **For shortcodes:**
+      - `shortcode` (`string`): The shortcode tag. **(Required)**
+      - `shortcode_attribute` (`string|array`): The attribute(s) containing the data. **(Required)**
+    - **For blocks:**
+      - `block_name` (`string`): The block name. **(Required)**
+      - `block_attribute` (`string|array`): The attribute(s) containing the data. **(Required)**
+
+  - **`pre_distribute_cb`** (`callable`):
+    A callback function that prepares the data on the source site before distribution. It should return extra data that is required for processing on the target site.
+
+  - **`post_distribute_cb`** (`callable`):
+    A callback function that processes the extra data on the target site and returns the updated ID to replace the source reference.
+
+---
+
+## Examples
+
+Below are several examples demonstrating how to use the `distributor_register_data` function in different scenarios.
+
+### 1. Handling an Image ID Stored in Post Meta
+
+In this example, we register an image ID stored in post meta under the key `example_image_id`.
+
+```php
+distributor_register_data( 'example_post_meta_data', array(
+	'location'   => 'post_meta',
+	'attributes' => array(
+		'meta_key' => 'example_image_id',
+	),
+	// Callback function to prepare the data before sending it from the source site.
+	// $image_id is the ID stored in post meta (example_image_id).
+	// $source_post_id is the ID of the post being distributed.
+	'pre_distribute_cb' => function ( $image_id, $source_post_id ) {
+		if ( ! $image_id ) {
+			return array();
+		}
+
+		$image_url = wp_get_attachment_image_url( $image_id, 'full' );
+
+		// Return extra data required for processing on the target site.
+		return array(
+			'image_url' => $image_url,
+		);
+	},
+	// Callback function to process the data before saving it on the target site.
+	// $extra_data is the data returned from pre_distribute_cb.
+	// $source_data is the original data stored in post meta.
+	// $post_data is the distributed post's data.
+	'post_distribute_cb' => function ( $extra_data, $source_data, $post_data ) {
+		if ( ! isset( $extra_data['image_url'] ) ) {
+			return;
+		}
+
+		$source_image_url = $extra_data['image_url'];
+		$source_image_id  = $source_data;
+
+		// Check if the media already exists on the target site. If yes, return its ID.
+		$image_id = Utils\get_attachment_id_by_original_data( $source_image_id, $source_image_url );
+
+		// If the media is not found, process and save it.
+		if ( ! $image_id ) {
+			$image_id = Utils\process_media( $source_image_url, 0, [] );
+			update_post_meta( $image_id, 'dt_original_media_id', $source_image_id );
+			update_post_meta( $image_id, 'dt_original_media_url', $source_image_url );
```

Comment:
Include slashing in example, as people will copy-paste.

Reply:
Handled in https://github.com/10up/distributor/pull/1302/commits/f8cf54a35b2609411fa6149bce6352f9deace7f7